src/platform_impl/linux/wayland/seat/pointer/mod.rs

```diff
@@ -58,7 +62,7 @@ impl WinitPointer {
             Some(cursor_icon) => cursor_icon,
             None => {
                 // Hide the cursor.
-                (*self.pointer).set_cursor(self.latest_serial.get(), None, 0, 0);
+                (*self.pointer).set_cursor(self.latest_enter_serial.get(), None, 0, 0);
```

Comment:
Could you add a note why the latest **enter** serial is required, so it won't get removed by accident.